### ISSUE TITLE
allow training with larger batch size

### ISSUE THREAD

**Issue by florisdesmedt** [fizyr/keras-retinanet#25]
in the code there are multiple asserts to only allow a batch size of 1. I tried starting the training on the COCO dataset, but with this restriction it would take a very large training time. 
What is the reason of this restriction?

**Comment by hgaiser** [comment]
It's not straightforward to allow for batch_size > 1.

One simple example, what do you do with two images, one of size 800x600 and one of size 600x600? The easiest solution is to pad the images with zeros such that they have the same shape as the largest image in a batch, which is what I intend to try. You could also group images by aspect ratio, which limits the amount of padding you'd need to do.

There are other layers and parts that assume a single image as input, because it makes implementation easier. Considering that the network works now, but trains slowly due to the batch size limitation, I have every intention of fixing this issue next.

tldr; it wasn't critical yet and simplified implementation, I hope to remove this limitation as soon as possible.

**Comment by florisdesmedt** [comment]
Maybe I can be of some help to get this feature integrated. It will require some time to get familiar with the code though.

The first problem that all images should be of the same size is a problem I have encountered a few times in the past. Generally padding with zeros (or copy border) seems like the easiest solution.
Layers that now expect a single image to be passed maybe a bit harder to solve, but I should look into the code for this to know for sure.

if I have created a solution that works, I will let you know and add it to the repository as a merge request.

**Comment by hgaiser** [comment]
I would gladly look at it and your help would be greatly appreciated :)

ps. I just created a Slack channel, #keras-retinanet on the Keras Slack team.

**Comment by hgaiser** [comment]
This issue is addressed in PR https://github.com/fizyr/keras-retinanet/pull/48 . I will close this when that PR gets merged.

**Comment by hgaiser** [comment]
This issue is addressed in PR https://github.com/fizyr/keras-retinanet/pull/48.
